dask/dataframe/io/parquet.py

```diff
@@ -223,11 +223,18 @@ def _write_fastparquet(df, path, write_index=None, append=False,
                     'Appended divisions overlapping with the previous ones.\n'
                     'New: {} | Previous: {}'.format(old_end, divisions[0]))
     else:
-        fmd = fastparquet.writer.make_metadata(df._meta,
-                                               object_encoding=object_encoding,
-                                               index_cols=index_cols,
-                                               ignore_columns=partition_on,
-                                               **kwargs)
+        # Get only arguments specified in the function
+        try:
+            from inspect import getfullargspec as getargspec
+        except ImportError:
+            from inspect import getargspec
+        spec = getargspec(fastparquet.writer.make_metadata)
+        safe_kwargs = {k: v for k, v in kwargs.items() if k in spec.args}
+        safe_kwargs['data'] = df._meta
+        safe_kwargs['object_encoding'] = object_encoding
+        safe_kwargs['index_cols'] = index_cols
+        safe_kwargs['ignore_columns'] = partition_on
+        fmd = fastparquet.writer.make_metadata(**safe_kwargs)
```

Comment:
Was a change for fastparquet necessary? From looking at the code, it looks like kwargs are only passed to `make_metadata`, which should then error if invalid kwargs are passed in. This is different from the pyarrow case you were fixing, where valid kwargs for the writer were erroneously being passed to the metadata writer.

Reply:
The added test `test_writing_parquet_with_unknown_kwargs` makes this function failed as the `unknown_key` is not removed. Indeed, this key is bogus but this will happen for any key we did not anticipate before and it is used in some other function. The code calling this function only pops some variables it uses internally but leaves all the others to this function. In general, see the reasoning in the comment for the test.